src/execution_plan.c

```diff
@@ -2342,6 +2369,7 @@ int FlatExecutionPlan_Register(FlatExecutionPlan* fep, ExecutionMode mode, void*
     RedisModule_Replicate(ctx, RG_INNER_REGISTER_COMMAND, "b", buff->buff, buff->size);
     RedisModule_FreeThreadSafeContext(ctx);
     Gears_BufferFree(buff);
+    FlatExecutionPlan_SetRegistered(fep);
```

Comment:
Should be done on FlatExecutionPlan_RegisterInternal which called also on slaves (and all the shards in the cluster ..)

Reply:
Also why you need a set function for it and not just set it, if you chose a function make if static ..